### PR TITLE
clean-up logging code, include kernel and particle info.

### PR DESCRIPTION
**Pull request by abhinavmuta**
The MultiStageEquation `__iter__` method may not be clean, any suggestions?

### CODE REVIEW

File: pysph/solver/application.py [hunk 1]
```diff
@@ -1236,6 +1237,24 @@ def _write_info(self, filename, **kw):
...
+        kernel_info = '%s(dim=%s)' % (kernel_name, solver.dim)
+
+        logger.info(
+            'No of particles:\n%s\n  %s\n%s' % (sep, particle_info, sep)
+        )
+        logger.info('Using kernel:\n%s\n  %s\n%s' % (sep, kernel_info, sep))
+        logger.info(
+            'Using integrator:\n%s\n  %s\n%s' % (sep, solver.integrator, sep)
+        )
+        logger.info('Using equations:\n%s\n%s\n%s' % (sep, eqn_info, sep))
```
Comment (42B) [inline pysph/solver/application.py]: Is there any reason why these `logger` calls aren't passing in the variables as shown in the Python docs?

```python
logger.info('Using equations:\n%s\n%s\n%s', sep, eqn_info, sep)  # ?
```

File: pysph/sph/equation.py [hunk 2]
```diff
@@ -871,3 +871,15 @@ def __repr__(self):
         return '%s(groups=[[\n%s\n    ]]\n)' % (
             name, '], [\n'.join(kw)
         )
+
+    def __iter__(self):
```
Comment (prabhuramachandran) [inline pysph/sph/equation.py]: Don't you just need to do return `iter(self.groups)` here and not have to define a `__next__`

File: pysph/sph/equation.py [hunk 3]
```diff
@@ -871,3 +871,15 @@ def __repr__(self):
...
+        self._cntr = 1
+        return self
+
+    def __next__(self):
+        # import pudb; pudb.set_trace()
```
Comment (prabhuramachandran) [inline pysph/sph/equation.py]: You can remove this.

File: pysph/solver/application.py [hunk 4]
```diff
@@ -1236,6 +1237,24 @@ def _write_info(self, filename, **kw):
...
+    def _log_solver_info(self, solver):
+        sep = '-'*70
+        eqn_info = '[\n' + ',\n'.join([str(e) for e in self.equations]) + '\n]'
+        particle_info = '\n  '.join(['%s: %d' % (p.name, len(p.gid)) for p in
+                                    solver.particles])
+
+        kernel_name = solver.kernel.__class__.__name__
+        kernel_info = '%s(dim=%s)' % (kernel_name, solver.dim)
+
+        logger.info(
```
Comment (prabhuramachandran) [inline pysph/solver/application.py]: Can use `self._message` for the particle information alone as it is often useful to see.  Also please provide information on the NNPS object used.

File: pysph/solver/application.py (hunk 1 above)
Comment (abhinavmuta) [reply pysph/solver/application.py]: Oh didn't know that, thanks!

File: pysph/sph/equation.py (hunk 2 above)
Comment (abhinavmuta) [reply pysph/sph/equation.py]: removed it, now we check if the `equation` is a list or not in the logging method, thanks!

File: pysph/solver/application.py (hunk 4 above)
Comment (abhinavmuta) [reply pysph/solver/application.py]: updated with nnps info, thanks!

File: pysph/solver/application.py (hunk 1 above)
Comment (42B) [reply pysph/solver/application.py]: Sorry, I should have included a link to the relevant docs.

* https://docs.python.org/3/library/logging.html#logging.debug

**Pull request merged**